applications/StructuralMechanicsApplication/custom_elements/small_displacement.cpp

```diff
@@ -24,368 +24,272 @@
+            rB( 2, i*3 + 2 ) = rDN_DX( i, 2 );
+            rB( 3, i*3     ) = rDN_DX( i, 1 );
+            rB( 3, i*3 + 1 ) = rDN_DX( i, 0 );
+            rB( 4, i*3 + 1 ) = rDN_DX( i, 2 );
+            rB( 4, i*3 + 2 ) = rDN_DX( i, 1 );
+            rB( 5, i*3     ) = rDN_DX( i, 2 );
+            rB( 5, i*3 + 2 ) = rDN_DX( i, 0 );
         }
+    }
 
-        if ( VOLUME_ACCELERATION.Key() == 0 )
-        {
-            KRATOS_ERROR <<"VOLUME_ACCELERATION has Key zero! (check if the application is correctly registered" << std::endl;
-        }
+    KRATOS_CATCH( "" )
+}
 
-        if ( THICKNESS.Key() == 0 )
-        {
-            KRATOS_ERROR <<"THICKNESS has Key zero! (check if the application is correctly registered" << std::endl;
-        }
+/***********************************************************************************/
+/***********************************************************************************/
 
-        // Verify that the dofs exist
-        for ( unsigned int i = 0; i < this->GetGeometry().size(); i++ )
-        {
-            if ( this->GetGeometry()[i].SolutionStepsDataHas( DISPLACEMENT ) == false )
-            {
-                KRATOS_ERROR <<"Missing variable DISPLACEMENT on node " << this->GetGeometry()[i].Id() << std::endl;
-            }
-
-            if ( this->GetGeometry()[i].HasDofFor( DISPLACEMENT_X ) == false || this->GetGeometry()[i].HasDofFor( DISPLACEMENT_Y ) == false || this->GetGeometry()[i].HasDofFor( DISPLACEMENT_Z ) == false )
-            {
-                KRATOS_ERROR <<"Missing one of the dofs for the variable DISPLACEMENT on node " << GetGeometry()[i].Id() << std::endl;
-            }
-        }
+Matrix SmallDisplacement::ComputeEquivalentF(const Vector& rStrainTensor)
+{
+    const unsigned int dim = GetGeometry().WorkingSpaceDimension();
+    Matrix F(dim,dim);
+    
+    if(dim == 2) {
+        F(0,0) = 1.0+rStrainTensor(0);  
+        F(0,1) = 0.5*rStrainTensor(2); 
+        F(1,0) = 0.5*rStrainTensor(2);   
+        F(1,1) = 1.0+rStrainTensor(1);
+    } else {
+        F(0,0) = 1.0+rStrainTensor(0);     
+        F(0,1) = 0.5*rStrainTensor(3); 
+        F(0,2) = 0.5*rStrainTensor(5);
+        F(1,0) = 0.5*rStrainTensor(3);   
+        F(1,1) = 1.0+rStrainTensor(1);   
+        F(1,2) = 0.5*rStrainTensor(4);
+        F(2,0) = 0.5*rStrainTensor(5);   
+        F(2,1) = 0.5*rStrainTensor(4); 
+        F(2,2) = 1.0+rStrainTensor(2);
+    }
+    
+    return F;
+}
 
-        // Verify that the constitutive law exists
-        if ( this->GetProperties().Has( CONSTITUTIVE_LAW ) == false )
-        {
-            KRATOS_ERROR << "Constitutive law not provided for property " << this->GetProperties().Id() << std::endl;
-        }
+/***********************************************************************************/
+/***********************************************************************************/
 
+int  SmallDisplacement::Check( const ProcessInfo& rCurrentProcessInfo )
```

Comment:
Why do you need this function? It is there in the baseclass already

Reply:
Just in case we want to add something, it calls the base class anyway

Reply:
ok, makes sense